pkg/network/setup/netstat.go

```diff
@@ -114,45 +114,61 @@ func (c *NetStat) UpdateStatus(vmi *v1.VirtualMachineInstance, domain *api.Domai
 	interfacesStatus = ifacesStatusFromGuestAgent(interfacesStatus, domain.Status.Interfaces)
 
 	primaryNetwork := netvmispec.LookupPodNetwork(vmi.Spec.Networks)
-	interfacesStatus = restorePodIfaceNameForPrimaryIface(interfacesStatus, primaryNetwork, vmi.Status.Interfaces)
+	if primaryNetwork != nil {
+		interfacesStatus = restorePrimaryIfaceStatus(interfacesStatus, vmi.Status.Interfaces, primaryNetwork.Name)
+	}
 	primaryInterfaceStatus, interfacesStatus := netvmispec.PopInterfaceByNetwork(interfacesStatus, primaryNetwork)
 	if primaryInterfaceStatus != nil {
 		interfacesStatus = append([]v1.VirtualMachineInstanceNetworkInterface{*primaryInterfaceStatus}, interfacesStatus...)
 	}
 
 	interfacesStatus = ifacesStatusFromMultus(interfacesStatus, multusStatusNetworksByName, vmiInterfacesSpecByName)
 
+	interfacesStatus = restorePodIfaceNames(interfacesStatus, vmi.Status.Interfaces)
```

Comment:
I don't follow, `restorePodIfaceNames` works on all interfaces (primary and secondaries).

Reply:
I meant closer to `restorePrimaryIfaceStatus` or the primary iface swap thing (placing it first in the slice).

Having `restorePodIfaceNames` handle all interfaces makes me wonder if it should merged with `restorePrimaryIfaceStatus`, WDYT?

Reply:
> I meant closer to `restorePrimaryIfaceStatus` or the primary iface swap thing (placing it first in the slice).
> 
> Having `restorePodIfaceNames` handle all interfaces makes me wonder if it should merged with `restorePrimaryIfaceStatus`, WDYT?

Could you please elaborate what would be the benefit of doing do?

`restorePrimaryIfaceStatus` restores the primary interface status in case it was reported by VMI controller, and is not yet reported by virt-handler.

`restorePrimaryIfaceStatus` restores the PodInterfaceNames of all interfaces.